### DISCUSSION TITLE
容器组件后续是否能够提供自定义呢？

### DISCUSSION THREAD

**Discussion by Channing-Huang** [jd-opensource/drip-form#191] [Ideas]
目前存在对容器组件的自定义需求，后续会开发吗？

**Comment by mengshang918** [comment]
1. 可以描述下需要的自定义功能
   方便后续自定义容器更好的扩展
3. 有些功能可以通过containerHoc实现
    ```
    type ContainerHoc = (
      FormItem: JSX.Element,
      formItemProps: {
        fieldKey: string
        error: string
        containerStyle: ContainerStyle
        type: string
        parentMode: string
        parentType: string
        // 是否是第一个表单
        isFirst: boolean
        uiProp: Record<string, any>
      }
    ) => JSX.Element
    ```
      目前generator的viewport拖拽功能也是通过containerHoc实现的。可以参考[generator form拖拽的containerHoc函数部分实现](https://github.com/JDFED/drip-form/blob/dev/packages/generator/src/components/Viewport/index.tsx)
   containerHoc文档在v0.8.0发布之后会进行重构完善，可关注#65
4. 后续会支持自定义容器嘛
    DripForm本质来讲是支持自定义容器的，但目前还有些前置需求需要开发。可以关注后续的roadmap
